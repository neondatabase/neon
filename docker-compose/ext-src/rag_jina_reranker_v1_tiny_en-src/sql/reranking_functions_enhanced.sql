-- Reranking function tests - single passage
SELECT rag_jina_reranker_v1_tiny_en.rerank_distance('the cat sat on the mat', 'the baboon played with the balloon');

SELECT rag_jina_reranker_v1_tiny_en.rerank_distance('the cat sat on the mat', 'the tanks fired at the buildings');

SELECT rag_jina_reranker_v1_tiny_en.rerank_distance('query about cats', 'information about felines');

SELECT rag_jina_reranker_v1_tiny_en.rerank_distance('', 'empty query test');

-- Reranking function tests - array of passages
SELECT rag_jina_reranker_v1_tiny_en.rerank_distance('the cat sat on the mat',
    ARRAY['the baboon played with the balloon', 'the tanks fired at the buildings']);

SELECT rag_jina_reranker_v1_tiny_en.rerank_distance('query about programming',
    ARRAY['Python is a programming language', 'Java is also a programming language', 'SQL is used for databases']);

SELECT rag_jina_reranker_v1_tiny_en.rerank_distance('empty array test', ARRAY[]::text[]);

-- Reranking score function tests - single passage
SELECT rag_jina_reranker_v1_tiny_en.rerank_score('the cat sat on the mat', 'the baboon played with the balloon');

SELECT rag_jina_reranker_v1_tiny_en.rerank_score('the cat sat on the mat', 'the tanks fired at the buildings');

SELECT rag_jina_reranker_v1_tiny_en.rerank_score('query about cats', 'information about felines');

SELECT rag_jina_reranker_v1_tiny_en.rerank_score('', 'empty query test');

-- Reranking score function tests - array of passages
SELECT rag_jina_reranker_v1_tiny_en.rerank_score('the cat sat on the mat',
    ARRAY['the baboon played with the balloon', 'the tanks fired at the buildings']);

SELECT rag_jina_reranker_v1_tiny_en.rerank_score('query about programming',
    ARRAY['Python is a programming language', 'Java is also a programming language', 'SQL is used for databases']);

SELECT rag_jina_reranker_v1_tiny_en.rerank_score('empty array test', ARRAY[]::text[]);
