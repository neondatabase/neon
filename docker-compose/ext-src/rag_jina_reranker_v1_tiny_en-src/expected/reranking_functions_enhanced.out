-- Reranking function tests - single passage
SELECT rag_jina_reranker_v1_tiny_en.rerank_distance('the cat sat on the mat', 'the baboon played with the balloon');
 rerank_distance 
-----------------
       0.8989152
(1 row)

SELECT rag_jina_reranker_v1_tiny_en.rerank_distance('the cat sat on the mat', 'the tanks fired at the buildings');
 rerank_distance 
-----------------
       1.3018152
(1 row)

SELECT rag_jina_reranker_v1_tiny_en.rerank_distance('query about cats', 'information about felines');
 rerank_distance 
-----------------
       1.3133051
(1 row)

SELECT rag_jina_reranker_v1_tiny_en.rerank_distance('', 'empty query test');
 rerank_distance 
-----------------
       0.7075559
(1 row)

-- Reranking function tests - array of passages
SELECT rag_jina_reranker_v1_tiny_en.rerank_distance('the cat sat on the mat',
    ARRAY['the baboon played with the balloon', 'the tanks fired at the buildings']);
    rerank_distance    
-----------------------
 {0.8989152,1.3018152}
(1 row)

SELECT rag_jina_reranker_v1_tiny_en.rerank_distance('query about programming',
    ARRAY['Python is a programming language', 'Java is also a programming language', 'SQL is used for databases']);
          rerank_distance           
------------------------------------
 {0.16591403,0.33475375,0.10132827}
(1 row)

SELECT rag_jina_reranker_v1_tiny_en.rerank_distance('empty array test', ARRAY[]::text[]);
 rerank_distance 
-----------------
 {}
(1 row)

-- Reranking score function tests - single passage
SELECT rag_jina_reranker_v1_tiny_en.rerank_score('the cat sat on the mat', 'the baboon played with the balloon');
 rerank_score 
--------------
   -0.8989152
(1 row)

SELECT rag_jina_reranker_v1_tiny_en.rerank_score('the cat sat on the mat', 'the tanks fired at the buildings');
 rerank_score 
--------------
   -1.3018152
(1 row)

SELECT rag_jina_reranker_v1_tiny_en.rerank_score('query about cats', 'information about felines');
 rerank_score 
--------------
   -1.3133051
(1 row)

SELECT rag_jina_reranker_v1_tiny_en.rerank_score('', 'empty query test');
 rerank_score 
--------------
   -0.7075559
(1 row)

-- Reranking score function tests - array of passages
SELECT rag_jina_reranker_v1_tiny_en.rerank_score('the cat sat on the mat',
    ARRAY['the baboon played with the balloon', 'the tanks fired at the buildings']);
      rerank_score       
-------------------------
 {-0.8989152,-1.3018152}
(1 row)

SELECT rag_jina_reranker_v1_tiny_en.rerank_score('query about programming',
    ARRAY['Python is a programming language', 'Java is also a programming language', 'SQL is used for databases']);
             rerank_score              
---------------------------------------
 {-0.16591403,-0.33475375,-0.10132827}
(1 row)

SELECT rag_jina_reranker_v1_tiny_en.rerank_score('empty array test', ARRAY[]::text[]);
 rerank_score 
--------------
 {}
(1 row)

